openmrs/omod/src/main/java/org/openmrs/projectbuendia/webservices/rest/PatientResource.java

```diff
@@ -369,38 +367,18 @@ private static void setLocation(Patient patient, String locationUuid) {
     }
 
     private SimpleObject searchInner(RequestContext requestContext) throws ResponseException {
-        List<Patient> patients = new ArrayList<>();
+        List<Patient> patients;
         String patientId = requestContext.getParameter("id");  // single patient by ID; overrides
-        // "q"
         if (patientId != null) {
             List<PatientIdentifierType> idTypes = new ArrayList<>();
             idTypes.add(DbUtil.getMsfIdentifierType());
             patients = patientService.getPatients(null, patientId, idTypes, true);
         } else {
-            String nameQuery = requestContext.getParameter("q");
-            patients = filterPatients(nameQuery, patientService.getAllPatients());
+            patients = patientService.getAllPatients();
```

Comment:
There isn't really a reason why this deletion is part of the PR, just that I hadn't got my head around how to do branching for this feature yet. I'll separate this commit out and send a seperate PR.

FWIW, refactoring sync is going to result in a number of breaking protocol changes. I was thinking I'd branch `dev` to `feature/sync-overhaul` in both the `projectbuendia/buendia` `projectbuendia/client` repos and then send PRs to `feature/sync-overhaul`. This allows for discussion on chunks of code along the way, whilst keeping work separate so as not to break other developers. Then at the end, we do a big merge back into `projectbuendia/buendia/dev`. Does that approach sound sensible to you?